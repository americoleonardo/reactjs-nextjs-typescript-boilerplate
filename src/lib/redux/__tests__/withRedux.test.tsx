import React from "react";
import { withRedux } from "../withRedux";
import { mount } from "enzyme";
import { useSelector } from "react-redux";

describe("<withRedux /> tests", () => {
  it("Mount component wirh Redux Store", async () => {
    expect.hasAssertions();

    const Test = () => {
      const state: any = useSelector((state) => state);

      expect(state).not.toBe(null);

      return <div>Test</div>;
    };

    const ComponentWithRedux = withRedux(Test, { ssr: true });

    const component = mount(<ComponentWithRedux />);

    expect(component.text()).toBe("Test");
    expect(component).toMatchSnapshot();
  });
});
