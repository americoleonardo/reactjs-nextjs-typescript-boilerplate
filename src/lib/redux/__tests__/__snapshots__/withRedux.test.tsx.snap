// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<withRedux /> tests Mount component wirh Redux Store 1`] = `
<withRedux(Test)>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Test>
      <div>
        Test
      </div>
    </Test>
  </Provider>
</withRedux(Test)>
`;
